packages/main/src/MultiInput.ts

```diff
@@ -13,81 +18,13 @@ import Input from "./Input.js";
 import MultiInputTemplate from "./generated/templates/MultiInputTemplate.lit.js";
 import styles from "./generated/themes/MultiInput.css.js";
 import Token from "./Token.js";
-import Tokenizer from "./Tokenizer.js";
+import Tokenizer, { ClipboardDataOperation } from "./Tokenizer.js";
 import Icon from "./Icon.js";
 import "@ui5/webcomponents-icons/dist/value-help.js";
 
-/**
- * @public
- */
-const metadata = {
-	tag: "ui5-multi-input",
-	properties: /** @lends sap.ui.webc.main.MultiInput.prototype */ {
-		/**
-		 * Determines whether a value help icon will be visualized in the end of the input.
-		 * Pressing the icon will fire <code>value-help-trigger</code> event.
-		 *
-		 * @type {boolean}
-		 * @defaultvalue false
-		 * @public
-		 */
-		showValueHelpIcon: {
-			type: Boolean,
-		},
-
-		/**
-		 * Indicates whether the tokenizer is expanded or collapsed(shows the n more label)
-		 * @private
-		 */
-		expandedTokenizer: {
-			type: Boolean,
-		},
-
-		/**
-		 * Indicates whether the tokenizer has tokens
-		 * @private
-		 */
-		tokenizerAvailable: {
-			type: Boolean,
-		},
-
-	},
-	slots: /** @lends sap.ui.webc.main.MultiInput.prototype */ {
-		/**
-		 * Defines the component tokens.
-		 *
-		 * @type {sap.ui.webc.main.IToken[]}
-		 * @slot
-		 * @public
-		 */
-		tokens: {
-			type: HTMLElement,
-		},
-	},
-	events: /** @lends sap.ui.webc.main.MultiInput.prototype */ {
-		/**
-		 * Fired when the value help icon is pressed
-		 * and F4 or ALT/OPTION + ARROW_UP/ARROW_DOWN keyboard keys are used.
-		 *
-		 * @event sap.ui.webc.main.MultiInput#value-help-trigger
-		 * @public
-		 */
-		"value-help-trigger": {},
-
-		/**
-		 * Fired when a token is about to be deleted.
-		 *
-		 * @event sap.ui.webc.main.MultiInput#token-delete
-		 * @param {HTMLElement} token deleted token.
-		 * @public
-		 */
-		"token-delete": {
-			detail: {
-				token: { type: HTMLElement },
-			},
-		},
-	},
-};
+type MultiInputTokenDeleteEventDetails = {
```

Comment:
Here in the type we can say that token has type Token and export that type.

```ts
type MultiInputTokenDeleteEventDetails = {
	token: Token;
}
```

Reply:
MultiInputTokenDeleteEventDetail not Details